src/tests/resistance/post_handshake/enormous_message.rs

```diff
@@ -149,6 +150,46 @@ async fn r004_t1_PROPOPSAL_PAYLOAD_send_a_huge_valid_msg() {
 
 #[tokio::test]
 #[allow(non_snake_case)]
-async fn r004_t2_XXX_send_a_huge_invalid_msg() {
-    // TODO(Rqnsom): implement
+async fn r004_t2_MSG_DIGEST_SKIP_send_a_huge_invalid_msg() {
+    // ZG-RESISTANCE-004
+    //
+    // Send a huge invalid message to the node and expect to lose connection.
+
+    // MsgDigestSkip imitation - expected message length is tag (2 bytes) + hash (32 bytes) = 34 bytes.
+    let mut msg = Tag::get_tag_str(&Tag::MsgDigestSkip).as_bytes().to_vec();
+    // Empirical value which the node won't reject.
+    let simple_data = vec![b'y'; 6_250_000];
+    msg.extend(simple_data);
+
+    // Spin up a node instance.
+    let target = TempDir::new().expect(ERR_TEMPDIR_NEW);
+    let mut node = Node::builder().build(target.path()).expect(ERR_NODE_BUILD);
+    node.start().await;
+
+    // Create a synthetic node.
+    let net_addr = node.net_addr().expect(ERR_NODE_ADDR);
+    let mut synthetic_node = get_handshaked_synth_node(net_addr).await;
+
+    synthetic_node
+        .unicast(net_addr, Payload::RawBytes(msg))
+        .expect(ERR_SYNTH_UNICAST);
+
+    // Clear the inbound queue.
+    while synthetic_node
+        .recv_message_timeout(Duration::from_millis(50))
+        .await
+        .is_ok()
+    {}
+
+    // Check that we are still receiving messages.
+    assert!(
+        !synthetic_node
+            .expect_message(&|m: &Payload| matches!(&m, _))
```

Comment:
There could be also used `recv_message_timeout` where you just give the timeout (it can be adjusted as now it's always 10 seconds - the default for `expect_message`) and there is no need to provide any matcher.

Reply:
Yes, I'm actually planning to write the `expect_message_timeout` function where one could specify the timeout for it
I would also use that in the C005 test where today by default we expect to get MsgOfINterest within 10 seconds, but we should probably use a lower value - like 3 seconds or smth,
wdyt?

Reply:
As for the `recv_message_timeout` suggestion, I tried it and the code gets longer... so I'm keeping `expect_message` until the `expect_message_timeout` is prepared

Reply:
Or maybe `expect_message` should be extended with optional parameter that would be timeout - it shouln't be very hard to change function calls with None, and we'd not create another function.

Reply:
Or at least I don't think so there are so many places to be changed :)

Reply:
> Or maybe expect_message should be extended with optional parameter that would be timeout

We could call the parameter
`override_default_timeout: Duration`

----
...now I'm getting crazy ideas like:

```rust
struct ExpectMsg {
   pub msg_closure: ...,
   pub timeout: Duration,
}

ExpectMsg:new(payload) -- which sets a default timeout
```
definition:
```rust
fn expect_message(&self, msg: ExpectMsg);
```
usage:
```rust
expect_message(&self, ExpectMsg::new(&|m: &Payload| matches!(&m, _)));
```

Reply:
`override_default_timeout: Option<Duration>` seems to be sufficient 😄

Reply:
> override_default_timeout: Option<Duration> seems to be sufficient 😄

haha yeah.
Let's bring this discussion on Slack - this is going to be done in a separate commit anyway